src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,304 @@
+  );
+}
+
+function createCategory(allRelationshipsObj) {
+  const newCategoryObj = {
+    category: intl.formatMessage({ id: 'ENTER_NEW_CATEGORY_NAME' }),
+    relationships: [],
+  };
+  allRelationshipsObj.push(newCategoryObj);
+  return allRelationshipsObj;
+}
+
+function updateCategoryName(
+  allRelationshipsObj,
+  currentCategoryName,
+  newCategoryName,
+) {
+  const targetRelationshipCategoryObj = get(
+    filter(
+      allRelationshipsObj,
+      currentRelationshipCategory =>
+        currentRelationshipCategory.category === currentCategoryName,
+    ),
+    '0',
+    {},
+  );
+  targetRelationshipCategoryObj.category = newCategoryName;
+  return recomposeAllRelationshipObjectFromModifiedCategoryObject(
+    allRelationshipsObj,
+    targetRelationshipCategoryObj,
+  );
+}
+
+function addRelationshipToCategory(
+  allRelationshipsObj,
+  currentCategory,
+  newRelationship,
+) {
+  const targetRelationshipCategoryObj = get(
+    filter(
+      allRelationshipsObj,
+      currentRelationshipCategory =>
+        currentRelationshipCategory.category === currentCategory,
+    ),
+    '0',
+    {},
+  );
+  const targetRelationshipArr =
+    targetRelationshipCategoryObj?.relationships;
+  targetRelationshipArr.push(newRelationship);
+  const modifiedArr = targetRelationshipArr;
+  const modifiedTargetCategoryObj = {
+    category: currentCategory,
+    relationships: modifiedArr,
+  };
+  return recomposeAllRelationshipObjectFromModifiedCategoryObject(
+    allRelationshipsObj,
+    modifiedTargetCategoryObj,
+  );
+}
+
+function renderSingleRelationship(
+  allRelationshipsObj,
+  currentCategory,
+  relationship,
+  onChange,
+) {
+  return (
+    <div style={{ marginLeft: 32, marginTop: 10 }}>
+      <TextInput
+        width={240}
+        schema={relationship}
+        onChange={newRelationshipName => {
+          onChange(
+            updateRelationshipInCategory(
+              allRelationshipsObj,
+              currentCategory,
+              relationship,
+              newRelationshipName,
+            ),
+          );
+        }}
+        value={relationship}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationshipsObj,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationshipsObj,
+  singleRelationshipObj,
+  onChange,
+) {
+  const category = singleRelationshipObj?.category;
+  const relationships = singleRelationshipObj?.relationships;
+  return (
+    <div style={{ marginTop: 10 }}>
+      <TextInput
+        width={240}
+        schema={category}
```

Comment:
`category` is just a string that is a user input. Why pass it as the `schema`?

Reply:
ahh shoot, this comment made me realize we are using the old form inputs here. everything in `src/components/inputs` is deprecated in favor of `src/components/fields/edit`. these components take a schema prop because they are set up to render form elements based on houston-supplied schemas (both the new and old version)

however, as seen here, it's possible to easily render a form element by "faking" a schema using only the schema properties necessary to get it looking the way you want. it's not clean and i'm not sure how i feel about it in the long haul but i think it's ok for mvp. 

better alternative is probably to use the `TextField` component straight from material-ui

Reply:
at some point i really need to migrate over the remaining things that are using `src/components/inputs` and delete this folder (there aren't many left but i never had time to get 'em all)

Reply:
Keeping as is for now. This will change in a future refactor.